### PR TITLE
Set up Mailhog

### PR DESCRIPTION
**Pull request by zamoose**
In lieu of Mailcatcher, [this discussion](https://github.com/Varying-Vagrant-Vagrants/VVV/issues/337) over on the VVV issue tracker turned me on to [Mailhog](https://github.com/mailhog/MailHog). MH seems to work more reliably and it has storage of email that persists daemon restarts. It installs easier and cleaner than Mailcatcher and seems to be more recently updated.

This PR installs and configures MH and sets up PHP and HHVM to use MH instead of system mail. It borrows concepts and some manifest structures from [here](https://github.com/geerlingguy/ansible-role-mailhog).

Addresses #86.

* [x] @zamoose
* [x] @markkelnar

### CODE REVIEW

File: provisioning/roles/common/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+d_workspace: /root
```
Comment (markkelnar) [inline provisioning/roles/common/defaults/main.yml]: Is this a leftover variable?  I don't see it used anywhere. d_workspace

**Comment by markkelnar** [comment]
This is a pretty cool admin panel to see emails sent.  I'll enjoy using this.

**Comment by markkelnar** [comment]
A side note: 
When I tried to provision this on an existing vagrant and the get_url failed pulling the tarball from https://github.com.  "Failed to validate the SSL certificate for github.com:443."

I destroyed the vagrant and provisioned a new one and it was fine and could not reproduce again.  I don't know if maybe there was an invalid/outdated CA certificate somewhere.

File: provisioning/roles/common/defaults/main.yml (hunk 1 above)
Comment (zamoose) [inline provisioning/roles/common/defaults/main.yml]: Nope, not used anywhere else. I think it's an artifact of the daemonize stuff I appropriated from upstream.

Probably safe to remove.

**Comment by zamoose** [comment]
Spurious variable removed from my branch. I'm going to go ahead and merge this.

**Pull request merged**